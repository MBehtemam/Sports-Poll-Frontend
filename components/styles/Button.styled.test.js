import React from "react";
import styled from "styled-components";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import "jest-styled-components";
import ButtonStyled from "./Button.styled";

describe("Test suits for button.styled", () => {
  it("should match with snapshot", () => {
    const tree = renderer.create(<ButtonStyled />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should has correct color for red button", () => {
    const tree = mount(<ButtonStyled color='red' />);
    expect(tree).toHaveStyleRule("color", "white");
  });
  it("should has correct color none button", () => {
    const tree = mount(<ButtonStyled />);
    expect(tree).toHaveStyleRule("color", "black");
  });
});
