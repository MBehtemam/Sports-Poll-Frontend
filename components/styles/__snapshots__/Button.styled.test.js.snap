// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suits for button.styled should match with snapshot 1`] = `
.c0 {
  border: none;
  padding: 15px;
  min-width: 300px;
  font-size: 20px;
  font-weight: 700;
  text-align: center;
  background-color: white;
  color: black;
}

<button
  className="c0"
/>
`;
